labs/parsers/python.py

```diff
@@ -0,0 +1,241 @@
+import ast
+import os
+
+
+class PythonFileParser(ast.NodeVisitor):
```

Comment:
Question: why do we need to parse the code AST? I just assumed we where sending code to the prompt and that's it, this seems a bit more complex.

Reply:
[Abstract Syntax Tree](https://docs.python.org/3/library/ast.html) allows you to parse Python code and is used as one of the methods for creating possible embeddins instead of breaking the file into random locations and losing the context and flow of the code